client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -153,6 +154,25 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 	);
 	const { setSelectedStyleVariation } = useDispatch( ONBOARD_STORE );
 
+	// When the theme query string parameter is present, automatically switch to previewing that theme (if it's a valid theme)
+	const themeFromQueryString = useQuery().get( 'theme' );
+	useEffect( () => {
+		if ( ! themeFromQueryString || ! allDesigns ) {
+			return;
+		}
+
+		const { generated: generatedDesigns, static: staticDesigns } = allDesigns;
+
+		const designs = [ ...generatedDesigns.designs, ...staticDesigns.designs ];
+
+		for ( const design of designs ) {
+			if ( design.slug === themeFromQueryString ) {
+				setSelectedDesign( design );
+				setIsPreviewingDesign( true );
+			}
+		}
```

Comment:
Some small notes on this implementation:
1. We're not breaking out of the loop when we find the expected design.
2. The for loop contains both our search logic and our resulting actions. (This one is way more of a preference.)

It may be easier to split this into separate steps using `find()` and an early exit condition.
```suggestion
		const designs = [ ...generatedDesigns.designs, ...staticDesigns.designs ];

		const requestedDesign = designs.find( ( design ) =>  design.slug === themeFromQueryString );
		if ( ! requestedDesign ) {
			return;
		}
		
		setSelectedDesign( requestedDesign );
		setIsPreviewingDesign( true );
```

Reply:
Much tidier, thanks!